src/QsCompiler/LlvmBindings/DebugInfo/DebugInfoBuilder.cs

```diff
@@ -226,7 +224,17 @@ public DIFile CreateFile(string? path)
         /// </returns>
         public DIFile CreateFile(string? fileName, string? directory)
         {
-            var handle = this.BuilderHandle.CreateFile(fileName, directory);
+            if (fileName == null)
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+
+            if (directory == null)
+            {
+                throw new ArgumentNullException(nameof(directory));
+            }
```

Comment:
Why not make `CreateFile` take in non-nullable strings as parameters? This looks like a source of bugs especially since the documentation for `CreateFile` explicitly says `fileName` and `directory` are allowed to be null.

Reply:
Ah, you're right, I missed the documentation there and the automerge kicked in before I saw your comment. Looking at the undelrying implementation, it's probably the intent to treat null as an empty string, so that's what I should be doing instead. `BuilderHandle.CreateFile` handles empty strings but does not check for null. This probably an oversight in the original code.